.gitlab/kernel_version_testing/system_probe.yml

```diff
@@ -259,3 +260,13 @@ kernel_matrix_testing_cleanup:
     - INSTANCE_IDS=$(aws ec2 describe-instances --filters "Name=tag:team,Values=ebpf-platform" "Name=tag:pipeline-id,Values=$CI_PIPELINE_ID" "Name=tag:managed-by,Values=pulumi" "Name=private-ip-address,Values=$INSTANCE_IPS" --output text --query 'Reservations[*].Instances[*].InstanceId' | tr '\n' ' ')
     - echo $INSTANCE_IDS
     - aws ec2 terminate-instances --instance-ids $INSTANCE_IDS
+
+kernel_matrix_testing_cleanup_on_failure:
+  extends:
+    - .kernel_matrix_testing_cleanup
+  when: on_failure
+
+kernel_matrix_testing_cleanup_on_success:
```

Comment:
Why the two jobs here?

Reply:
You can use `always` to always run this job

Reply:
That is exactly what I was looking for.